users.go

```diff
@@ -441,17 +446,17 @@ func (s *UsersService) UnblockUser(user int, options ...OptionFunc) error {
 	}
 
 	resp, err := s.client.Do(req, nil)
-	if err != nil {
+	if err != nil && resp == nil {
```

Comment:
If a method returns an error, you should not try to use any of the other returned values. As the error indicates things didn't go as planned, you cannot trust what the other values contain.

So please do not use `resp` if the same call returning `resp` also returned an error. This is common and best practice as far as I know...

Reply:
Haven't tested this out myself, but I'm assuming that the error from the GitLab API is less than useful (as it is the case on some endpoints). Without asserting the resp is nil here I'm not sure what a cleaner way to leverage the `switch` below for informative errors.

Of course if the gilab api is returning useful error messages and they are already getting propagated through the client these switch cases can probably be deleted.

Reply:
Looking at the rest of the package, I would almost say it's much more inline to just remove the whole switch statement here. We don't do that in any other location and the error returned is already parsed and put into a custom helpful format in the `Do` method.

We do try to return the received response so users can choose to use/see the raw response (mainly for debugging and investigating in case of issues). So using the `resp` it in this case (even when you got an error) doesn't seem to be a bad choice. But still, do we really want this? Do we want to sugarcoat all the errors we return with custom messages for each individual call there is?

I think we shouldn't do that... What we now return is something like `GET https://gitlab.com/api/v3/some/path: 403 error message from API` (https://github.com/xanzy/go-gitlab/blob/master/gitlab.go#L608-L612) I think that is just fine right? If people using this package want to sugarcoat that a bit more because the want it to be returned to the user or a UI, it's easy to do that in their own package right?